compiler/AST/AggregateType.cpp

```diff
@@ -2283,7 +2283,8 @@ void AggregateType::fieldToArg(FnSymbol*              fn,
 
         DefExpr*    defPoint = field->defPoint;
         const char* name     = field->name;
-        ArgSymbol*  arg      = new ArgSymbol(INTENT_IN, name, dtUnknown);
+        IntentTag intent     = field->hasFlag(FLAG_CONST) ? INTENT_CONST_IN : INTENT_IN;
+        ArgSymbol*  arg      = new ArgSymbol(intent, name, dtUnknown);
```

Comment:
I'm not sure this will work. IIRC an `owned` variable can be transferred out of only when it is not `const`. (Is the compiler just not checking that for default initializers?) Also, it worries me that this change applies to the default initializer but not to user initializers. Why do we feel the need to change this for default initializers but not user initializers?

Reply:
After a discussion offline, we reverted this change and try to determine whether we are calling a default initializer with `in` intents at the call site. New commits reflect that change.